csharp_dotnetcore/3.Welcome-Users/Startup.cs

```diff
@@ -0,0 +1,135 @@
+                    // is restarted, anything stored in memory will be gone.
+                    IStorage dataStore = new MemoryStorage();
+
+                    // For production bots use Azure Blob, or Azure CosmosDB storage provides
+                    // as seen below. To include any of the Azure based storage providers,
+                    // add the Microsoft.Bot.Builder.Azure  Nuget package to your solution. That package is found at:
+                    //      https://www.nuget.org/packages/Microsoft.Bot.Builder.Azure/
+
+                    // IStorage Store = new Microsoft.Bot.Builder.Azure.AzureBlobStorage("AzureBlobConnectionString", "containerName");
+
+                    // var convoState = new ConversationState(dataStore);
+                    // options.State.Add(new ConversationState(dataStore));
+                    options.State.Add(new UserState(dataStore));
+
+                    // Add State to BotStateSet Middleware (that require auto-save)
+                    // The BotStateSet Middleware forces state storage to auto-save when the Bot is complete processing the message.
+                    // Note: Developers may choose not to add all the State providers to this Middleware if save is not required.
+                    // var stateSet = new BotStateSet(options.State.ToArray());
+                    options.Middleware.Add(new BotStateSet(options.State.ToArray()));
+                });
+
+                // register 'object' (classes) into the runtime services collection
+                services.AddSingleton<WelcomeUserStateAccessors>(sp =>
+                {
+                    var options = sp.GetRequiredService<IOptions<BotFrameworkOptions>>().Value;
+                    if (options == null)
+                    {
+                        throw new InvalidOperationException("BotFrameworkOptions must be configured prior to setting up the State Accessors");
+                    }
+
+                    var userState = options.State.OfType<UserState>().FirstOrDefault();
+                    if (userState == null)
+                    {
+                        throw new InvalidOperationException("ConversationState must be defined and added before adding conversation-scoped state accessors.");
+                    }
+
+                    // Create Custom State Property Accessors
+                    // State Property Accessors enable components to read and write individual properties, without having to
+                    // pass the entire State object.
+                    var accessors = new WelcomeUserStateAccessors
+                    {
+                        DidBotWelcomedUser = userState.CreateProperty<bool>("DidBotWelcomeState"),
+                    };
+
+                    return accessors;
+                });
+            }
+            catch (Exception e)
+            {
+                Console.WriteLine(e);
+                throw;
+            }
+        }
+
+        /// <summary>
```

Comment:
I would remove the comments here totally. @daveta & I checked to see what the ASP.Net Core team did for their samples, and this method wasn't commented in the ones we wchecked.

Reply:
it is a public method - needs comments